climada_petals/entity/exposures/osm_dataloader.py

```diff
@@ -0,0 +1,346 @@
+        return f'[out:json][timeout:180];(nwr{self.condition}{self.area};(._;>;););out;'
+
+    def _assemble_from_relations(self, result):
+        """
+        pick out those nodes and ways from result instance that belong to
+        relations. Assemble relations into gdfs. Keep track of which nodes
+        and ways have been "used up" already
+
+        Parameters
+        ---------
+        result : overpy.Overpass result object
+
+        Returns
+        -------
+        nodes_taken : list
+            node-ids that have been used to construct relations.
+            Not "available" anymore for further constructions
+        ways_taken : list
+            way-ids that have been used to construct relations.
+            Not "available" anymore for further constructions
+        gdf_rels : gpd.GeoDataFrame
+            gdf with relations that were assembled from result object
+        """
+
+        nodes_taken = []
+        ways_taken = []
+        data_geom = []
+        data_id = []
+        data_tags = []
+
+        for relation in result.relations:
+            data_tags.append(relation.tags)
+            data_id.append(relation.id)
+            roles = []
+            relationways = []
+
+            for way in relation.members:
+                relationways.append(way.ref)
+                roles.append(way.role)
+
+            ways_taken.append(relationways)
+
+            nodes_taken_mp, gdf_polys = self._assemble_from_ways(
+                result, relationways, closed_lines_are_polys=True)
+
+            nodes_taken.append(nodes_taken_mp)
+
+            gdf_polys['role'] = roles
+
+            # separate relationways into inner, outer polygons and linestrings,
+            # combine them.
+
+            # step 1: polygons to multipolygons
+            inner_mp = shapely.geometry.MultiPolygon(
+                gdf_polys.geometry[(gdf_polys.geometry.type=='Polygon') &
+                                   (gdf_polys.role=='inner')].values)
+            outer_mp = shapely.geometry.MultiPolygon(
+                gdf_polys.geometry[(gdf_polys.geometry.type=='Polygon') &
+                                   (gdf_polys.role=='outer')].values)
+
+            # step 2: poly from lines --> multiline --> line --> polygon
+            lines = gdf_polys.geometry[
+                (gdf_polys.geometry.type=='LineString')].values
+            if len(lines) > 0:
+                poly = shapely.geometry.Polygon(
+                    shapely.ops.linemerge(shapely.geometry.MultiLineString(lines)))
+            else:
+                poly = shapely.geometry.Polygon([])
+
+            # step 3: combine to one multipoly
+            multipoly = shapely.ops.unary_union([outer_mp - inner_mp, poly])
+            data_geom.append(multipoly)
+
+            if multipoly.area == 0:
+                LOGGER.info('Empty geometry encountered.')
+
+        gdf_rels =  gpd.GeoDataFrame(
+            data={'osm_id': data_id,'geometry': data_geom, 'tags':data_tags})
```

Comment:
This raises a warning. GeoDataFrames must be initialized with explicit `geometry` parameter. Also, the CRS is missing.